docs/spec/relayer-old/ChannelHandlers.tla

```diff
@@ -0,0 +1,172 @@
+ channel datagrams
+ ***************************************************************************)
+
+EXTENDS Naturals, FiniteSets
+
+ChannelIDs == {"chanAtoB", "chanBtoA"}
+nullChannelID == "none"
+
+ChannelStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN", "CLOSED"}
+ChannelOrder == {"ORDERED", "UNORDERED"}
+
+(***************************************************************************
+ Channel helper operators
+ ***************************************************************************)
+
+\* get the channel ID of the channel end at the connection end of chainID
+GetChannelID(chainID) ==
+    IF chainID = "chainA"
+    THEN "chanAtoB"
+    ELSE IF chainID = "chainB"
+         THEN "chanBtoA"
+         ELSE nullChannelID
+         
+\* get the channel ID of the channel end at chainID's counterparty chain
+GetCounterpartyChannelID(chainID) ==
+    IF chainID = "chainA"
+    THEN "chanBtoA"
+    ELSE IF chainID = "chainB"
+         THEN "chanAtoB"
+         ELSE nullChannelID          
+
+(***************************************************************************
+ Channel datagram handlers
+ ***************************************************************************)
+ 
+\* Handle "ChanOpenInit" datagrams
+HandleChanOpenInit(chainID, chain, datagrams) ==
+    \* get "ChanOpenInit" datagrams, with a valid channel ID
+    LET chanOpenInitDgrs == {dgr \in datagrams : 
+                            /\ dgr.type = "ChanOpenInit"
+                            /\ dgr.channelID = GetChannelID(chainID)} IN
+    
+    \* if there are valid "ChanOpenInit" datagrams and the connection is not "UNINIT", 
+    \* create a new channel end and update the chain
+    IF chanOpenInitDgrs /= {} /\ chain.connectionEnd.state /= "UNINIT"
+    THEN LET chanOpenInitDgr == CHOOSE dgr \in chanOpenInitDgrs : TRUE IN
+         LET chanOpenInitChannelEnd == [
+             state |-> "INIT",
+             channelID |-> chanOpenInitDgr.channelID,
+             counterpartyChannelID |-> chanOpenInitDgr.counterpartyChannelID
+         ] IN 
+         LET chanOpenInitConnectionEnd == [
+             chain.connectionEnd EXCEPT !.channelEnd = chanOpenInitChannelEnd
+         ] IN
+         LET chanOpenInitChain == [
+            chain EXCEPT !.connectionEnd = chanOpenInitConnectionEnd            
+         ] IN
+        
+         \* TODO: check here if channel is already in INIT?
+         \* TODO: when handling packets later on, set nextSequenceRecv and nextSequenceSend to 1
+         chanOpenInitChain
+    \* otherwise, do not update the chain     
+    ELSE chain
+
+\* Handle "ChanOpenTry" datagrams
+HandleChanOpenTry(chainID, chain, datagrams) ==
+    \* get "ChanOpenTry" datagrams, with a valid channel ID
+    LET chanOpenTryDgrs == {dgr \in datagrams : 
+                            /\ dgr.type = "ChanOpenTry"
+                            /\ dgr.channelID = GetChannelID(chainID)
+                            /\ dgr.proofHeight \in chain.counterpartyClientHeights} IN
+    
```

Comment:
Yes, that's correct, the `datagrams` is a set and there is no order in which the messages are received currently. However, the way in which the operator `ChannelUpdate` in Chain.tla is defined, the `openInit` is processed before `openTry`, since the `HandleChanOpenInit` operator is applied first, and then `HandleChanOpenTry` is applied next (and `HandleChanOpenConfirm`, `HandleChanOpenAck`, in that order). 

I suppose that when we specify the event-based relayer, we would switch from sets to queues to store the incoming datagrams. This would induce a minor change in the handlers, exactly in these lines.